CapuaEnvironment/ExecutionUnit/ExecutionUnit.py

```diff
@@ -246,37 +231,37 @@ def getRegisterValue(self, registerCode=None):
         :param registerCode: int, from 0b00 to 0b11
         :return:
         """
-        if registerCode == REGISTER_A:
+        if registerCode == REGISTERS["A"]:
             register = self.A
-        elif registerCode == REGISTER_B:
+        elif registerCode == REGISTERS["B"]:
             register = self.B
-        elif registerCode == REGISTER_C:
+        elif registerCode == REGISTERS["C"]:
             register = self.C
-        elif registerCode == REGISTER_D:
+        elif registerCode == REGISTERS["D"]:
             register = self.D
-        elif registerCode == REGISTER_E:
+        elif registerCode == REGISTERS["E"]:
             register = self.E
-        elif registerCode == REGISTER_F:
+        elif registerCode == REGISTERS["F"]:
             register = self.F
-        elif registerCode == REGISTER_G:
+        elif registerCode == REGISTERS["G"]:
             register = self.G
-        elif registerCode == REGISTER_A2:
+        elif registerCode == REGISTERS["A2"]:
             register = self.A2
-        elif registerCode == REGISTER_B2:
+        elif registerCode == REGISTERS["B2"]:
             register = self.B2
-        elif registerCode == REGISTER_C2:
+        elif registerCode == REGISTERS["C2"]:
             register = self.C2
-        elif registerCode == REGISTER_D2:
+        elif registerCode == REGISTERS["D2"]:
             register = self.D2
-        elif registerCode == REGISTER_E2:
+        elif registerCode == REGISTERS["E2"]:
             register = self.E2
-        elif registerCode == REGISTER_F2:
+        elif registerCode == REGISTERS["F2"]:
             register = self.F2
-        elif registerCode == REGISTER_G2:
+        elif registerCode == REGISTERS["G2"]:
             register = self.G2
-        elif registerCode == REGISTER_S:
+        elif registerCode == REGISTERS["S"]:
             register = self.S
-        elif registerCode == REGISTER_S2:
+        elif registerCode == REGISTERS["S2"]:
```

Comment:
So I have ideas that would continue simplifying this. But it would result in some hackery and potentially changing a lot of the core. This is where I'm going to ask @pg-cse to weigh in before changing this further.

Effectively the above code performs a linear search of the keys inside `REGISTERS`, once a matchin  value is found, a hardcoded object ref (ie `self.S`) is assigned to `register`. 

It would be easier to read if the linear search was just a linear search (IMHO), like so:
```python3
register = None
for k,v in REGISTERS.items():
   if registerCode == v:
      register = self.registers[k]
if not register:
   raise CoreError("Invalid register code packed inside instruction at address {}".format(/*I can't remember where this is*/))
```
Note: better names would be used for `k` and `v`.

But that would require putting all registers in a dictionnary, effectively requiring the change of ALL uses of registers in the core. Big change for small gain.

While there is a hacky way to get around adding a `REGISTERS` dict inside the core, namely this:
```python3
self.__dict__[key]
```
That would be one of the most hilarious hacks I've come up with and isn't something I'd call kosher. I will see if there is anything online about using this. SO I went and dug. Its regarded as hackery depending on who you ask BUT when accessing a variable, `self.G2` for instance, is the same as `self.__dict__['G2']`. Because under the hood, a python object is effectively a dictionary but without the `code object`( Just means the source code isn't referred to by the instance ). 

To summarize, we can turn this bit, and other bits like this,  into linear searches to reduce the amount of code present and address the registers via the underlying dictionary of the python object. Whether this is a worthwhile change, is something I'd like @pg-cse 's opinion on.

Reply:
Again, I know doing the dict thing for the registers feels like a good idea. However, that part of the code had originally been profiled for performance and the "ugliness" of the current code was the result of that performance check. It was (back then we only had 4 registers) what was the fastest way of running that code when I tested it 3 years ago. With the extended registers this could not be the truth anymore.

Allowing this PR into main repo is more involving and will require some performance profiling on my part. The fact that this PR also changes much more than just the assembler is a bit of concern to me. I will need some time to evaluate this one.